tests/has/devfile_source.go

```diff
@@ -134,8 +134,13 @@ var _ = framework.HASSuiteDescribe("[test_id:01] devfile source", Label("has"),
 	})
 
 	It("Gitops Repository should not be deleted when component gets deleted", func() {
-		componentName2 := fmt.Sprintf(QuarkusComponentName+"-%s", util.GenerateRandomString(10))
-		component2, err := framework.HasController.CreateComponentFromStub(compDetected, componentName2, testNamespace, "", applicationName)
+		comp2Detected := appservice.ComponentDetectionDescription{}
+
+		for _, comp2Detected = range cdq.Status.ComponentDetected {
+			comp2Detected.ComponentStub.ComponentName = "java-quarkus2"
+		}
+		component2Name := fmt.Sprintf(QuarkusComponentName+"-%s", util.GenerateRandomString(10))
+		component2, err := framework.HasController.CreateComponentFromStub(comp2Detected, component2Name, testNamespace, "", applicationName)
```

Comment:
After introducing component health checks into our tests, this test started failing on "component2" creation:
```
Component create failed: application already has a component with name java-quarkus
```

so what I did here is just an update to the component detection description's componentName field so the second component name differs from the previous one and the creation doesn't fail.

I think this test was working fine before because application-service controller didn't had a chance to reconcile the "component2" CR before the test deleted it.